src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -340,7 +353,14 @@ end = struct
             Either.First
               (Num_accounts
                  (len, MT.get_inner_hash_at_addr_exn mt content_root_addr) )
+        | What_child_hashes a ->
+            let ledger_depth = MT.depth mt in
+            let addresses = intermediate_range ledger_depth a subtree_depth in
+            let get_hash a = MT.get_inner_hash_at_addr_exn mt a in
+            let hashes = List.map addresses ~f:get_hash in
```

Comment:
You should use `Or_error.try_with` to catch and handle any errors, like the previous version of the code did.